server/settings/main.py

```diff
@@ -169,6 +169,46 @@ class VersionStartCategoryModel(BaseSettingsModel):
     )
 
 
+class EnvironmentReplacementModel(BaseSettingsModel):
+    environment_key: str = SettingsField("", title="Enviroment variable")
+    pattern: str = SettingsField("", title="Pattern")
+    replacement: str = SettingsField("", title="Replacement")
+
+
+class FilterFarmEnvironmentModel(BaseSettingsModel):
+    _layout = "expanded"
+
+    hosts: list[str] = SettingsField(
+        default_factory=list,
+        title="Host names"
+    )
+
+    task_types: list[str] = SettingsField(
+        default_factory=list,
+        title="Task types",
+        enum_resolver=task_types_enum
+    )
+
+    task_names: list[str] = SettingsField(
+        default_factory=list,
+        title="Task names"
+    )
+
+    folders: list[str] = SettingsField(
```

Comment:
I still don't think this is good idea...

Reply:
Changed to `folder_paths`. 

Maybe you want to modify different env vars for `shots` vs `characters`?